wsmaster/codenvy-hosted-api-audit/src/main/java/com/codenvy/api/audit/server/AuditManager.java

```diff
@@ -0,0 +1,174 @@
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.inject.Inject;
+import javax.inject.Singleton;
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+import static com.google.common.io.Files.createTempDir;
+import static java.util.Collections.emptyList;
+
+/**
+ * Facade for audit report related operations.
+ *
+ * @author Igor Vinokur
+ */
+@Singleton
+public class AuditManager {
+
+    private static final Logger LOG = LoggerFactory.getLogger(AuditManager.class);
+
+    private final JpaUserDao            userDao;
+    private final WorkspaceManager      workspaceManager;
+    private final PermissionsManager    permissionsManager;
+    private final CodenvyLicenseManager licenseManager;
+    private final AuditReportPrinter    reportPrinter;
+    private final UserManager           userManager;
+
+    private AtomicBoolean inProgress = new AtomicBoolean(false);
+
+    @Inject
+    public AuditManager(UserManager userManager,
+                        JpaUserDao userDao,
+                        WorkspaceManager workspaceManager,
+                        PermissionsManager permissionsManager,
+                        CodenvyLicenseManager licenseManager,
+                        AuditReportPrinter reportPrinter) {
+        this.userManager = userManager;
+        this.userDao = userDao;
+        this.workspaceManager = workspaceManager;
+        this.permissionsManager = permissionsManager;
+        this.licenseManager = licenseManager;
+        this.reportPrinter = reportPrinter;
+    }
+
+    /**
+     * Generates audit report.
+     *
+     * @return path of audit report file
+     * @throws ServerException
+     *         if an error occurs
+     * @throws ConflictException
+     *         if generating report is being run by other user
+     */
+    public Path generateAuditReport() throws ServerException, ConflictException {
+        if (!inProgress.compareAndSet(false, true)) {
+            throw new ConflictException("This command is being run by other user");
+        }
+
+        Path auditReport = null;
+        try {
+            String dateTime = new SimpleDateFormat("dd-MM-yyyy_hh:mm:ss").format(new Date());
+            auditReport = Paths.get(createTempDir().getAbsolutePath(), "/report_" + dateTime + ".txt");
+            Files.createFile(auditReport);
+
+            CodenvyLicense license = null;
+            try {
+                license = licenseManager.load();
+            } catch (LicenseException ignored) {
+                //Continue printing report without license info
+            }
+            reportPrinter.printHeader(auditReport, userDao.getTotalCount(), license);
```

Comment:
Use https://github.com/eclipse/che/blob/master/wsmaster/che-core-api-user/src/main/java/org/eclipse/che/api/user/server/UserManager.java#L238 instead of method in user dao